# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/EarlyHintPreloader.h
# Commit: 8575cd92e1d8
# Full Hash: 8575cd92e1d8cd783e6cca3f6feadef11f0c4cc8
# Author: Manuel Bucher <manuel@mozilla.com>
# Date: 2022-12-03 09:24:59
# Regressor Bug: 1771867
# File Overlap Count: 1
# Description:
#   Bug 1771867 - Early Hints Phase 2 - Part 8: Add list of open channels from OngoingEarlyHints to RedirectToRealChannelArgs r=necko-reviewers,valentin,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161179
# ==============================================================================

diff -r ce8f0a8fa688 -r 8575cd92e1d8 netwerk/protocol/http/EarlyHintPreloader.h
--- a/netwerk/protocol/http/EarlyHintPreloader.h	Fri Dec 02 16:15:55 2022 +0000
+++ b/netwerk/protocol/http/EarlyHintPreloader.h	Fri Dec 02 16:15:56 2022 +0000
@@ -22,6 +22,7 @@
 namespace mozilla::net {
 
 class EarlyHintPreloader;
+class EarlyHintConnectArgs;
 struct LinkHeader;
 
 // class keeping track of all ongoing early hints
@@ -38,9 +39,17 @@
 
   void CancelAllOngoingPreloads();
 
+  // registers all channels and returns the ids
+  void RegisterLinksAndGetConnectArgs(
+      nsTArray<EarlyHintConnectArgs>& aOutLinks);
+
  private:
   ~OngoingEarlyHints() = default;
   nsRefPtrHashtable<PreloadHashKey, EarlyHintPreloader> mOngoingPreloads;
+
+  // keep track of all preloads in the order they were specified in the early
+  // hint header
+  nsTArray<EarlyHintConnectArgs> mLinks;
 };
 
 class EarlyHintPreloader final : public nsIStreamListener,